crates/typst/src/image.rs

```diff
@@ -306,55 +306,100 @@ fn decode_svg(
     Ok(Rc::new(DecodedImage::Svg(tree)))
 }
 
+/// A font family and its variants.
+#[derive(Clone)]
+struct FontData {
+    /// The usvg-compatible font family name.
+    usvg_family: EcoString,
+    /// The font variants included in the family.
+    fonts: EcoVec<Font>,
+}
+
 /// Discover and load the fonts referenced by an SVG.
 fn load_svg_fonts(
     tree: &usvg::Tree,
     loader: Tracked<dyn SvgFontLoader + '_>,
 ) -> fontdb::Database {
     let mut fontdb = fontdb::Database::new();
-    let mut referenced = BTreeMap::<EcoString, Option<EcoString>>::new();
+    let mut font_cache = BTreeMap::<EcoString, FontData>::new();
+    let mut loaded = BTreeMap::<EcoString, ()>::new();
 
-    // Loads a font family by its Typst name and returns its usvg-compatible
-    // name.
-    let mut load = |family: &str| -> Option<EcoString> {
+    // Loads a font family by its Typst name and returns its data.
+    let mut load = |family: &str| -> Option<FontData> {
         let family = EcoString::from(family.trim()).to_lowercase();
-        if let Some(success) = referenced.get(&family) {
-            return success.clone();
+        if let Some(success) = font_cache.get(&family) {
+            return Some(success.clone());
+        }
+
+        let fonts = loader.load(&family);
+        let usvg_family = fonts.iter().find_map(|font| {
+            font.find_name(ttf_parser::name_id::TYPOGRAPHIC_FAMILY)
+                .or_else(|| font.find_name(ttf_parser::name_id::FAMILY))
+                .map(Into::<EcoString>::into)
+        });
+
+        let font_data = FontData { usvg_family: usvg_family?, fonts };
```

Comment:
If a font doesn't have a family, it would be loaded and loaded over again, failing over and over again. I admit that this will probably not happen in practice though. Still, I'd typically just cache the option instead.